### ISSUE TITLE
NaN parameter gradients from fractional powers at zero (vdiff, JAM, SSO) — and no repo-wide d/dparam test

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#663]
Found by a re-audit of the July 2026 ECHAM+RRTMGP review against `dev` @ `f6d1bcd`. Parent: #614.

## What's wrong

The pattern `where(cond, power(maximum(x, 0), frac), 0)` has the classic JAX failure: at `x = 0` the fractional-power derivative is ∞ and the `where` cotangent multiplies `0 · ∞ = NaN`. The fix is to double-`where` the *base*.

Campaign PR 2 fixed the two `echam_1m` sites and added a parameter-gradient regression test, but **the repo-wide sweep for other sites was never done** — it was carried into #614's re-audit list and is still open. The sweep has now been done and the class survives in three places, two of them demonstrated.

## Confirmed NaN, with probes

**`jcm/physics/vertical_diffusion/tte_tke/tke_budget.py:208-215`** — `echam_tke_source_update`:
```
prev_tke=0      tke=0.01  d/dprev_tke=nan
prev_tke=1e-12  tke=0.01  d/dprev_tke=0.0
```
`prev_tke` is `state.tke` (`tte_tke/vertical_diffusion.py:231`) and `VerticalDiffusionData.zeros` initialises `tke=jnp.zeros(...)` (`vertical_diffusion_types.py:324`) — so **exactly-zero TKE is the cold-start value**, and any multi-step reverse-mode gradient of the ECHAM stack NaNs on step 1 through the vertical-diffusion carry.

**`jcm/physics/aerosol/jam/activation/arg_term.py:96-103`** and **`jam/ice_nucleation/ice_term.py:63-70`** — `w = sqrt(maximum(tke_factor*tke, 0))` then `maximum(w, w_min)`:
```
tke=0     w=0.1  d/dtke=nan
d/d(tke_factor) at tke=0 = nan
```
Here the NaN reaches a **calibratable parameter**: `tke_factor` sets the activation and ice-nucleation updraft, i.e. the sensitivity of the entire aerosol→CDNC and aerosol→ICNC pathway.

## Same class, not probed

- `gravity_waves/sso/lott_miller.py:614` — `zalfa = sqrt(max(pstab*zdz2, 0))` where `zdz2 ∝ ptau`, and `ptau` is set to exactly `0.0` by the `zero_branch` arm two lines later (`:626`), which supplies the 0 cotangent.
- `diagnostics/aerocom.py:864` — `w_act = 0.7*sqrt(max(2*tke, 0))` inside `where(has_cloud, take, 0.0)` (`:872`); the composition probes NaN at tke=0.
- Lower reachability: `clouds/lohmann_2m/scheme.py:237`, `aerosol/jam/chemistry/aqueous.py:149` (degenerate discriminant).

Verified safe: `echam_1m.py:881` (base strictly positive), `flux_tendencies.py:162`, `deposition_freezing.py:692` (floored at 1e-30, finite slope), all `maximum(x, 1e-30)`-style sites.

## Why the tests miss it

∂/∂state often survives in the same column while ∂/∂parameter is NaN — which is exactly why the existing checks pass. The only repo-wide guard, `physics/echam/gradient_finiteness_test.py`, differentiates w.r.t. the **solar constant** only; no parameter struct is a differentiation target anywhere outside the two smooth-gradient test files.

## Fix

1. Mechanical double-where at each site:
   ```python
   has = x > 0
   y = jnp.where(has, jnp.power(jnp.where(has, x, 1.0), frac), 0.0)
   ```
   Same pattern as the existing `zxrp1_base` fix at `echam_1m.py:892`.
2. **Close the actual gap:** extend `gradient_finiteness_test.py` to differentiate w.r.t. every float leaf of every term's `nnx.Param` struct, on both an aquaplanet-isothermal and a moist state. That is the repo-wide ∂/∂param test whose absence let this class survive the campaign, and it is the part that stops it recurring.

## Refs

Maintainability review B.0 and its unconfirmed residual sweep. Cross-ref #262 — related but do **not** merge: every skipped check in #262 is SPEEDY-family, and none of the sites above is SPEEDY.
